vignettes/introduction-to-partition.Rmd

```diff
@@ -69,56 +71,90 @@ df <- simulate_block_data(
+correlation_subset <- corr(baxter_otu[, 1:200])
+ggcorrplot::ggcorrplot(correlation_subset, hc.order = TRUE) + ggplot2::theme_void()
 ```
 
-partition takes a data frame and a threshold--the minimum amount of information a reduced variable must explain to be created--and reduces the data to as few variables as possible. The threshold prevents information loss: each reduced variable must still explain at least that much.
+# Reducing data with partition
+
+Because there are many more features (OTUs) in these data than rows (patients), it's useful to reduce data to meaningful subsets to use in statistical modeling. The primary function in partition is `partition()`. `partition()` takes a data frame and a threshold--the minimum amount of information a reduced variable must explain to be created--and reduces the data to as few variables as possible. The threshold prevents information loss: each reduced variable must still explain at least greater than or equal to the information threshold.
 
 ```{r}
-prt <- partition(df, threshold = .5)
+prt <- partition(baxter_otu, threshold = .5)
 
 prt
 ```
 
-partition creates a `tibble` with the newly reduced variables:
+For the microbiome data, `partition()` reduced 158 of the OTUs to 63 reduced variables. The other 1,076 OTUs were not reduced because doing so would have removed too much information from the data. `partition()` creates a `tibble` with the newly reduced variables, as well as any variables that were not reduced, which you can get with `partition_scores()`:
 
 ```{r}
 partition_scores(prt)
 ```
 
-Each variable in the original data set maps to one reduced variable; in this partition, there are four reduced variables consisting of two to three variables each, as well as five of the original variables that did not get reduced because too much information would have been lost.
+In comparison, PCA produces a data set of the same dimensions of the original data, with each component representing multiple variables, and with variables usually mapping to several components. While components are organized by their informativeness (the first component explains the most variance), no original features are retained. 
 
 ```{r}
-plot_ncluster(prt) +
+pca <- prcomp(baxter_otu)
+
+# print the results more neatly
+tibble::as_tibble(pca$x)
+```
+
+Notably, these approaches can be easily combined (see below).
+
+# The partition algorithm
+
+partition uses an approach called Direct-Measure-Reduce to create agglomerative (bottom-up) partitions that maintain the user-specified minimum level of information. Each variable starts as an individual cluster; candidate reduced variables are assessed by making sure this level of information is kept. The reduced variables are also interpretable: original variables map to one and only one variable in the reduced data set. partition is flexible, as well: how variables are selected to reduce, how information loss is measured, and the way data is reduced can all be customized. 
```

Comment:
partition uses an approach called Direct-Measure-Reduce to create agglomerative (bottom-up) partitions that capture the user-specified minimum level of information. Each variable starts as an individual partition subset, and candidate partition subsets are assessed by verifying that the minimum information is captured in the reduced variable. Reduced variables are easily interpretable because original variables map to one and only one variable in the reduced data set. The partition software is flexible and customizable in the way features are agglomerated, information is measured, and data are reduced.